src/main/java/com/aws/greengrass/lifecyclemanager/ShellRunner.java

```diff
@@ -40,7 +40,7 @@ class Default implements ShellRunner {
         DeviceConfiguration deviceConfiguration;
 
         @Override
-        public synchronized Exec setup(String note, String command, GreengrassService onBehalfOf) throws IOException {
+        public Exec setup(String note, String command, GreengrassService onBehalfOf) throws IOException {
```

Comment:
From the code changes I noticed a few other places that don't lock, and they either make a simple call to get a value or calls a function that locks, but why does this function now not need to lock?

Reply:
Never needed any lock. There's no state associated with ShellRunnner which this method does (or could) change.